primitives/utils/src/metrics.rs

```diff
@@ -0,0 +1,52 @@
+
+// Copyright 2020 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Metering primitives and globals
+
+use lazy_static::lazy_static;
+use prometheus::{
+	Registry, Error as PrometheusError, Opts,
+	core::{
+		GenericGaugeVec, AtomicU64,
+		GenericGauge, GenericCounter
+	}
+};
+
+lazy_static! {
+	pub static ref TOKIO_THREADS_TOTAL: GenericCounter<AtomicU64> = GenericCounter::new(
```

Comment:
Thinking a bit more about the two tokio metrics, we still support different runtimes, right? If so these would be exposed even when using e.g. async-std.

If that is a problem we could as well derive similar information from sysinfo in regards to the current amount of threads (https://docs.rs/sysinfo/0.12.0/sysinfo/struct.Process.html).

Reply:
> we still support different runtimes, right?

We don't at the moment. And this is the least worrying place to change if we did.

> If that is a problem we could as well derive similar information from sysinfo in regards to the current amount of threads

How? Where? I only see a list of `Pid`, which means child processes. Threads of the process are currently pulled in via `procinfo` – and because not everything runs on the tokio runtime the systems threads counter has a very different number. Until we have unified that, I'd prefer to have both.

Reply:
Not sure `tasks` is used instead of threads or children: https://docs.rs/sysinfo/0.12.0/sysinfo/struct.Process.html

Either way, let's keep as is, given that it is not very intrusive.